### PR TITLE
Refactoring part 1: Fix some annoying linter warnings in API code

### PR DESCRIPTION
**Pull request by kkovaletp**
Related to #953 

- var, const, func names, 
- constants of reusable strings, 
- removed unneeded `if`, 
- replaced deprecated imports

### CODE REVIEW

File: api/tools/tools.go
Comment (kkovaletp) [inline api/tools/tools.go:1]: @googollee, I see that this file is basically empty:
- it has 2 strange comments at the beginning
- then I see the error for the package definition: `No packages found for open file /Users/kkoval/Documents/Repos/photoview/api/tools/tools.go.`
- finally, the import returns the error: `import "github.com/99designs/gqlgen" is a program, not an importable package`

Is it OK to have this file here 'as is'? Is it used? How does it work?

PS: I've added the comment in line 10 just to have the file in the PR. I'll revert it before the merge.

**Comment by kkovaletp** [comment]
@googollee, another question is about https://github.com/photoview/photoview/blob/master/api/graphql/generated.go and https://github.com/photoview/photoview/blob/master/api/graphql/models/generated.go files:
I found that they contain the most linter errors and were updated last time 5 months ago. Since that, we've updated the version of `gqlgen` and other dependencies.

Does it make sense to regenerate these files?
How to do that?
In general, at which time does it make sense to regenerate them?
Is there a way to instruct `gqlgen` to generate a better compliant code with fewer code smells?

**Comment by kkovaletp** [comment]
I'll create another PR with long functions refactoring, as this type of change doesn't directly depend on the other changes here, and this PR is already long enough

File: api/scanner/scanner_tasks/processing_tasks/process_photo_task.go [hunk 1]
```diff
@@ -96,11 +96,9 @@ func (t ProcessPhotoTask) ProcessMedia(ctx scanner_task.TaskContext, mediaData *
 	if origURL == nil {
```
Comment (googollee) [inline api/scanner/scanner_tasks/processing_tasks/process_photo_task.go:96]: It's better to move the definition of `photoDimensions` in line 58 just before this `if` condition.

File: api/tools/tools.go
Comment (googollee) [reply api/tools/tools.go:1]: This is marked `gqlgen` as a dependency in `go.mod`.

`go:build tools` means this file won't be compiled if the building flag `tools` isn't provided. `+build` is the same but for legacy style.

We should keep it if it doesn't break any thing.

File: api/graphql/resolvers/faces.go [hunk 2]
```diff
@@ -11,6 +11,11 @@ import (
 	"gorm.io/gorm"
 )
 
+const faceDetectorNotInitialized = "face detector not initialized"
+const faceGroupIDisQuestion = "face_group_id = ?"
+const mediaAlbumIDinQuestion = "media.album_id IN (?)"
+const imageFacesIDinQuestion = "image_faces.id IN (?)"
```
[the remark this answers is not in the document]
Comment (kkovaletp) [reply api/graphql/resolvers/faces.go:17]: Yes, like for other literals, repeated in the code 3 times or more. Maybe you could propose better names for these consts? Is there a better practice for coding in such cases, when you have to repeat the same SQL part many times?

[the remark this answers is not in the document]
Comment (googollee) [reply api/graphql/resolvers/faces.go:17]: Well. I don't have any good idea immediately. Let's keep those consts.

File: api/graphql/models/actions/timeline_actions.go [hunk 3]
```diff
@@ -8,6 +8,8 @@ import (
 	"gorm.io/gorm"
 )
 
+const albumsTitleASC = "albums.title ASC"
```
[the remark this answers is not in the document]
Comment (kkovaletp) [reply api/graphql/models/actions/timeline_actions.go]: Done

File: api/graphql/models/album_test.go [hunk 4]
```diff
@@ -8,12 +8,16 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+const photosPath = "/photos"
+const photosChild1Path = "/photos/child1"
+const photosChild1SubchildPath = "/photos/child1/subchild"
```
[the remark this answers is not in the document]
Comment (kkovaletp) [reply api/graphql/models/album_test.go]: Done

File: api/graphql/models/media_test.go [hunk 5]
```diff
@@ -8,6 +8,8 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+const imageJpeg = "image/jpeg"
```
[the remark this answers is not in the document]
Comment (kkovaletp) [reply api/graphql/models/media_test.go]: Done

File: api/graphql/resolvers/faces.go [hunk 6]
```diff
@@ -11,6 +11,11 @@ import (
 	"gorm.io/gorm"
 )
 
+const faceDetectorNotInitialized = "face detector not initialized"
```
[the remark this answers is not in the document]
Comment (kkovaletp) [reply api/graphql/resolvers/faces.go]: Done

File: api/scanner/exif/exif_parser_internal.go [hunk 7]
```diff
@@ -17,12 +17,16 @@ import (
 // internalExifParser is an exif parser that parses the media without the use of external tools
 type internalExifParser struct{}
 
+const exifTagReturnedNull = "exif tag returned null"
```
[the remark this answers is not in the document]
Comment (kkovaletp) [reply api/scanner/exif/exif_parser_internal.go]: Done

[the remark this answers is not in the document]
Comment (kkovaletp) [reply api/scanner/exif/exif_parser_internal.go]: Done

File: api/scanner/scanner_tasks/processing_tasks/process_photo_task.go (hunk 1 above)
Comment (kkovaletp) [reply api/scanner/scanner_tasks/processing_tasks/process_photo_task.go:96]: Done

**Review by googollee — APPROVED** [review]

**Pull request merged**